appdatabase/database.go

```diff
@@ -198,11 +188,144 @@ func MigrateEnsUsernames(db *sql.DB) error {
+	// efficient manner
+	// Extract useful information from the tx blob
+	// Dropped access_list, need a separate table for it
+	// Already there chain_id, type, hash
+	// Dropped v, r, s because I see no way to be useful as BLOBs
+	// Added BIGINT values as clamped 64 INT because we can't use 128 bits blobs/strings for int arithmetics
+	// _clamped64 prefix indicate clamped 64 bits INT values might be useful for queries (sorting, filtering ...)
+	newColumnsAndIndexSetup := `
+		ALTER TABLE transfers ADD COLUMN status INT;
+		ALTER TABLE transfers ADD COLUMN receipt_type INT;
+		ALTER TABLE transfers ADD COLUMN cumulative_gas_used INT;
+		ALTER TABLE transfers ADD COLUMN contract_address TEXT;
+		ALTER TABLE transfers ADD COLUMN gas_used INT;
+		ALTER TABLE transfers ADD COLUMN tx_index INT;
+
+		ALTER TABLE transfers ADD COLUMN protected BOOLEAN;
+		ALTER TABLE transfers ADD COLUMN gas_limit UNSIGNED INT;
+		ALTER TABLE transfers ADD COLUMN gas_price_clamped64 INT;
+		ALTER TABLE transfers ADD COLUMN gas_tip_cap_clamped64 INT;
+		ALTER TABLE transfers ADD COLUMN gas_fee_cap_clamped64 INT;
+		ALTER TABLE transfers ADD COLUMN amount_padded128hex CHAR(32);
+		ALTER TABLE transfers ADD COLUMN account_nonce INT;
+		ALTER TABLE transfers ADD COLUMN size INT;
+
+		CREATE INDEX transfers_status ON transfers (status);`
+
+	rowIndex := 0
+	mightHaveRows := true
+
+	_, err := sqlTx.Exec(newColumnsAndIndexSetup)
+	if err != nil {
+		return err
+	}
+
+	for mightHaveRows {
+		var chainID uint64
+		var hash common.Hash
+		var address common.Address
+
+		rows, err := sqlTx.Query(`SELECT hash, address, network_id, tx, receipt FROM transfers WHERE tx IS NOT NULL OR receipt IS NOT NULL LIMIT ? OFFSET ?`, batchSize, rowIndex)
+		if err != nil {
+			return err
+		}
+
+		curProcessed := 0
+		for rows.Next() {
+			tx := &types.Transaction{}
+			r := &types.Receipt{}
+
+			// Scan row data into the transaction and receipt objects
+			nullableTx := sqlite.JSONBlob{Data: tx}
+			nullableR := sqlite.JSONBlob{Data: r}
+			err = rows.Scan(&hash, &address, &chainID, &nullableTx, &nullableR)
+			if err != nil {
+				rows.Close()
+				return err
+			}
+
+			var currentRow []interface{}
+			// Check if the receipt is not null before transferring the receipt data
+			if nullableR.Valid {
+				currentRow = append(currentRow, r.Status, r.Type, r.CumulativeGasUsed, r.ContractAddress, r.GasUsed, r.TransactionIndex)
+			} else {
+				for i := 0; i < 6; i++ {
+					currentRow = append(currentRow, nil)
+				}
+			}
+
+			if nullableTx.Valid {
+				gasPrice := sqlite.BigIntToClampedInt64(tx.GasPrice())
+				gasTipCap := sqlite.BigIntToClampedInt64(tx.GasTipCap())
+				gasFeeCap := sqlite.BigIntToClampedInt64(tx.GasFeeCap())
+				value := sqlite.BigIntToPadded128BitsStr(tx.Value())
+
+				currentRow = append(currentRow, tx.Protected(), tx.Gas(), gasPrice, gasTipCap, gasFeeCap, value, tx.Nonce(), int64(tx.Size()))
```

Comment:
There's a couple other fields I think we won't be able to reconstruct
```
	if t.BaseGasFees != "" {
		baseFee := new(big.Int)
		baseFee.SetString(t.BaseGasFees[2:], 16)
		tip := t.Transaction.EffectiveGasTipValue(baseFee)

		view.EffectiveTip = (*hexutil.Big)(tip)
		price := new(big.Int).Add(baseFee, tip)
		view.EffectiveGasPrice = (*hexutil.Big)(price)
	}
```

`	view.Input = hexutil.Bytes(t.Transaction.Data())`

```
		if t.Transaction.To() != nil {
			view.To = *t.Transaction.To()
		}
```

Reply:
I had a quick look, and this seems to be a task for the extensive DB reorganization we talked about recently that won't require data migration.